features/google/src/components/activitiesOverTimeStory/activitiesOverTimeStory.jsx

```diff
@@ -0,0 +1,190 @@
+                    <p>{i18n.t("activitiesOverTimeStory:tab.arrow.right")}</p>
+                    <img
+                        src="./images/angle-right.svg"
+                        alt="arrow-right"
+                        className="space-left"
+                    />
+                </button>
+            ) : (
+                <div className="arrow filler"></div>
+            )}
+        </div>
+    );
+};
+
+export const ActivitiesOverTimeStorySummary = ({ activitiesOverTime }) => {
+    return (
+        <div className="render-summary">
+            <p className="highlighted-number">
+                {activitiesOverTime.total.toLocaleString("de-DE")}
+            </p>
+            {i18n.t("activitiesOverTimeStory:summary", {
+                number_activities: activitiesOverTime.total,
+            })}
+            <p className="source">{i18n.t("common:source.your.google.data")}</p>
+        </div>
+    );
+};
+
+export const ActivitiesOverTimeStoryDetails = ({ activitiesOverTime }) => {
+    const yearRange = fillMissingArrayValues(
+        Object.keys(activitiesOverTime.values)
+    );
+
+    const yearlyTotals = yearRange.map((year) => {
+        return {
+            title: year.toString().substring(2, 4),
+            value: activitiesOverTime.values[year]?.total || 0,
+        };
+    });
+
+    const [selectedYear, setSelectedYear] = useState(
+        yearRange[yearRange.length - 1]
+    );
+
+    const monthlyTotals = monthsAbbreviation.map((month, index) => {
+        return {
+            title: month,
+            value: activitiesOverTime.values[selectedYear]?.values[index] || 0,
+        };
+    });
+
+    const tabData = [
+        {
+            id: "total",
+            translation: i18n.t("activitiesOverTimeStory:tab.total"),
+            barData: yearlyTotals,
+            barWidth: 6,
+            barChartLegendText: i18n.t(
+                "activitiesOverTimeStory:tab.events.total",
+                {
+                    number_events: activitiesOverTime.total,
+                }
+            ),
+            barValueColor: null,
+            datePicker: (
+                <DatePicker
+                    year={selectedYear}
+                    yearRange={yearRange}
+                    onYearChange={setSelectedYear}
+                />
+            ),
+            belowChart: selectedYear,
+        },
+        {
+            id: "yearly",
+            translation: i18n.t("activitiesOverTimeStory:tab.year"),
+            barData: monthlyTotals,
+            barWidth: 22,
+            barChartLegendText: i18n.t(
+                "activitiesOverTimeStory:tab.events.yearly",
+                {
+                    number_events:
+                        activitiesOverTime.values[selectedYear]?.total || 0,
+                }
+            ),
+            barValueColor: "white",
+            datePicker: <div className="datepicker-filler" />,
+            belowChart: i18n.t("common:total.years"),
+        },
+    ];
```

Comment:
The issue with this data structure is that it's, on one hand, quite similar to the equivalent in fbImport, on the other, it's an array instead of a hash; keys there have turned into id's here. That kinda prevents from using some kind of common class, or refactoring to a common component. I guess it makes sense for a start, anyway.

Reply:
We should refactor this to a component as it's improved and use it in fb. Yet as said putting something like a story to poly-look needs praparation